RELEASE_NOTES.md

```diff
@@ -41,6 +41,9 @@ v1.0.0-alpha.X
   provide more context when using it as a starting point for an
   OpenAssetIO integration.
 
+- Support Python 3.11 for building the Python API.
+  [#863](https://github.com/OpenAssetIO/OpenAssetIO/issues/863)
+
```

Comment:
[fix] Technically, this change doesn't fully support 3.11. It will allow the project to build, but until we update the tests as per #683, and have 3.11 in the CI matrix, then we can't fully claiming support.  Can we reword this, and the commit title to be just about the `pybind` version update?

Reply:
Done, thanks.

Reply:
Thanks, can we update `RELEASE_NOTES` too please to mention the Pybind version change in the  supplied `conanfile` instead of that we support Python 11 - as we can't really say that till we fix the tests. Our build instructions cover how to test your build.

Reply:
Awesome, thanks! Don't worry, its a somewhat fiddly process until you get used to it :)